[2025-03-07 13:56:39,644][src.utils.utils][INFO] - Disabling python warnings! <cfg.extras.ignore_warnings=True>
[2025-03-07 13:56:39,644][src.utils.utils][INFO] - Enforcing tags! <cfg.extras.enforce_tags=True>
[2025-03-07 13:56:39,647][src.utils.utils][INFO] - Printing config tree with Rich! <cfg.extras.print_config=True>
[2025-03-07 13:56:39,647][src.utils.rich_utils][WARNING] - Field 'data' not found in config. Skipping 'data' config printing...
[2025-03-07 13:56:39,647][src.utils.rich_utils][WARNING] - Field 'model' not found in config. Skipping 'model' config printing...
[2025-03-07 13:56:39,647][src.utils.rich_utils][WARNING] - Field 'callbacks' not found in config. Skipping 'callbacks' config printing...
[2025-03-07 13:56:39,661][__main__][INFO] - Instantiating trainer <src.models.trainer_image.ImageTrainer>
[2025-03-07 13:56:39,663][src.utils.utils][ERROR] - 
Traceback (most recent call last):
  File "/opt/miniconda3/envs/mi3/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 92, in _call_target
    return _target_(*args, **kwargs)
TypeError: Can't instantiate abstract class ImageTrainer with abstract methods on_stage_end, on_stage_start

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/wangzh/Desktop/ShanghaiTech/Spring_2025/ICCP2025/mitsuba3-workflow/src/utils/utils.py", line 68, in wrap
    metric_dict, object_dict = task_func(cfg=cfg)
  File "/Users/wangzh/Desktop/ShanghaiTech/Spring_2025/ICCP2025/mitsuba3-workflow/src/train.py", line 28, in train
    trainer: MitsubaTrainer = hydra.utils.instantiate(cfg.trainer)
  File "/opt/miniconda3/envs/mi3/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 226, in instantiate
    return instantiate_node(
  File "/opt/miniconda3/envs/mi3/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 347, in instantiate_node
    return _call_target(_target_, partial, args, kwargs, full_key)
  File "/opt/miniconda3/envs/mi3/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 97, in _call_target
    raise InstantiationException(msg) from e
hydra.errors.InstantiationException: Error in call to target 'src.models.trainer_image.ImageTrainer':
TypeError("Can't instantiate abstract class ImageTrainer with abstract methods on_stage_end, on_stage_start")
full_key: trainer
[2025-03-07 13:56:39,664][src.utils.utils][INFO] - Output dir: /Users/wangzh/Desktop/ShanghaiTech/Spring_2025/ICCP2025/mitsuba3-workflow/logs/fit_image_test/runs/2025-03-07_13-56-39
